File: movie-service/src/main/java/com/movies/movieserver/movie/Movie.java
package com.movies.movieserver.movie;

import java.util.List;

public record Movie(
        int id,
        String title,
        String original_title,
        String overview,
        String release_date,
        String poster_path,
        String backdrop_path,
        double popularity,
        double vote_average,
        int vote_count,
        List<Integer> genre_ids,
        String original_language,
        boolean adult,
        boolean video) {
}

File: frontend/src/types/movie.ts

export interface MovieData {
  id: number;
  title: string;
  original_title: string;
  overview: string;
  release_date: string;
  poster_path: string;
  backdrop_path: string;
  popularity: number;
  vote_average: number;
  vote_count: number;
  genre_ids: number[];
  original_language: string;
  adult: boolean;
  video: boolean;
  isFavorite: boolean;
}